src/ORM/Association.php

```diff
@@ -514,13 +515,21 @@ public function attachTo(Query $query, array $options = [])
     {
         $target = $this->target();
         $joinType = empty($options['joinType']) ? $this->joinType() : $options['joinType'];
+
+        $table = $target->table();
+        if (($target->defaultConnectionName() !== $this->source()->defaultConnectionName())
+            && ($this->source()->connection()->supportsCrossWith($target->connection()))
```

Comment:
This seems like it is duplicating logic that should part of `supportsCrossWith()`.